README.md

```diff
@@ -1,39 +1,67 @@
+```sh
+docker image build -t backstage-plugin-risk-scorecard-backend .
+```
+To run the docker image, run:
 
-### Environment variables
-**SOPS_AGE_KEY** the age key used to encrypt and decrypt the RiSc Analyses. The public key of this asymmetric pair will have to be added and used to encrypt all RiScs that the backend is supposed to decrypt.
-The private key of the pair will be used to decrypt. This key is the SOPS_AGE_KEY environment variable.
+```sh
+docker-compose run --build local
+```
 
-How to make one?
+# Setup from scratch: Preconditions and environment variables
+## Github App
+The Github App is used to issue oauth2 tokens in order to communicate with the Github repositories where the risc analyses are stored.
+The permissions that are asked for is set in the GithubHelper. Permissions for the access token requested is these:
 
-``` shell
-brew install age-keygen
-age-keygen -o key.txt
+```json
+{
+  "repositories": ["<the repo that the risc belongs to>"],
+  "permissions": "contents:write,pull_requests:write,statuses:read"
+}
 ```
 
-This will create a file with the private key, and output the public key.
-Set the secret key as your environment variable, and the public key in the key groups of .sops.yaml.
+### New Github App
+Set up a Github App in your organisation and give the following permissions:
+Contents: read/write, commit statuses: read/write, pull requests: read/write, and personal access token requests: read/write.
 
-**GITHUB_PRIVATE_KEY_BASE64_ENCODED** is a private key to the GitHub App used to access the repositories of the organisation. 
-The key is used to sign jwt-tokens. Because of line ending encoding is hard, we base64 encode the private PEM key and set this as our environment variable.
+**GITHUB_INSTALLATION_ID**: Locate your application, for example by using this link https://github.com/organizations/<your-org>/settings/installations. Click your application, and find the installation id in the url.
 
-How to set it?
-- Go to your GitHub App settings, and scroll down to "Private keys"
-- Create one if you don't already have one
+**GITHUB_APP_ID**: When you have created and installed your github app you can locate it here: https://github.com/organizations/<your-org>/settings/apps/<your-app>.
+
+**GITHUB_PRIVATE_KEY_BASE64_ENCODED** is a private key generated for the service, and base64 encoded because formats are hard. Generate a new private key, and base64 encode the PEM-key.
 
 ``` shell
 echo "<  -----BEGIN RSA PRIVATE KEY----- ... -----END RSA PRIVATE KEY----->" | base64
 ```
 
-And set this as the environment variable.
+## OAuth Server
+We use the backend of the backstage plugin as our issuer/oauth server.
+
+**ISSUER_URI**: In our case the issuer_uri has been the backstage application "https://kv-ros-backstage-245zlcbrnq-lz.a.run.app/api/auth".
```

Comment:
Hadde det vært greit å skrive hva urien er når man skal kjøre det lokalt? Evt. ha en egen seksjon om miljøvariabler el. for lokal utvikling?

Reply:
La til en kommentar i .env-fila, er det ok? eller skal jeg legge til i readme også? :)